code/modules/projectiles/ammunition/boxes.dm

```diff
@@ -46,7 +46,7 @@
 	name = "ammo box (12mm)"
 	icon_state = "9mm"
 	origin_tech = "combat=2"
-	ammo_type = "/obj/item/ammo_casing/a12mm"
+	ammo_type = "/obj/item/ammo_casing/a12mm/assault"
```

Comment:
I think it'd be better to make a child of `/obj/item/ammo_storage/box/c12mm` and change *its* ammo_type to your new bullets, so that it would still be possible for regular 12mm bullets to be spawned.
Of course, you would also have to alter the protolathe recipe to point to your new ammo box.